src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -23,16 +24,18 @@ public class Autogram {
     private final DriverDetector driverDetector;
     private final boolean shouldDisplayVisualizationError;
     private final Integer slotId;
+    private final TSPSource tspSource;
 
-    public Autogram(UI ui, boolean shouldDisplayVisualizationError , DriverDetector driverDetector) {
-        this(ui, shouldDisplayVisualizationError, driverDetector, -1);
+    public Autogram(UI ui, boolean shouldDisplayVisualizationError , DriverDetector driverDetector, TSPSource tspSource) {
```

Comment:
Tu sa nam to zacina kopit a imho by vlastne mal do Autogramu vliezt nejaky settings. To by potom riesilo aj problem, ze potrebujeme restart.